### DISCUSSION TITLE
Close ContextFlyout on contained button click

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#13307] [Q&A] [answered]
I have a treeview with a context flyout.
The flyout has a textbox and a button:

      <TreeView.ContextFlyout>
        <Flyout FlyoutPresenterClasses="RoundedCorners">
          <Menu >
            <StackPanel>            
              <TextBox Watermark="###.###.###.###" Text="{Binding IPAddress, Mode=TwoWay}"/>
              <Button Content="Connect"  Command="{Binding ConnectCommand}"/>
            </StackPanel>
          </Menu>
        </Flyout>
      </TreeView.ContextFlyout>

The flyout opens on a right-click in the treeview.
I would like to close the flyout if the button is clicked.
How do I get to the flyout instance to call Hide() on it?
Is there another way to accomplish this?

**Comment by maxkatz6** [comment, marked as answer]
If you get access to TreeView instance, you can do `TreeView.ContextFlyout.Hide()` I believe.

**Reply by wgraaf** [reply]
Used TreeView.ContextFlyout.Hide() in code-behind for now.

**Comment by timunie** [comment]
I guess you want to close that flyout from the command? If so, you need some way of Interaction. 

Ideas:
- use RxUI Interacion: https://github.com/timunie/Avalonia.Samples/tree/feature/MVVM_Dialogs/src/Avalonia.Samples/ViewInteraction/MvvmDialogSample
- use DialogManager: https://github.com/timunie/Avalonia.Samples/tree/feature/MVVM_Dialogs/src/Avalonia.Samples/ViewInteraction/AdvancedMvvmDialogSample
- Use a kind of Messenger: https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/messenger

**Comment by ValerCheck** [comment]
Thanks to this thread I was able to find another solution for my needs without code-behind which looks also nice.

If someone was looking for example with interactions, here it is:

```cs
<UserControl
    xmlns:i="using:Avalonia.Xaml.Interactivity"
    xmlns:ib="using:Avalonia.Xaml.Interactions.Core"
    <TreeView.ContextFlyout>
        <Flyout FlyoutPresenterClasses="RoundedCorners">
            <Menu >
                <StackPanel>            
                    <Button Content="Connect"  Command="{Binding ConnectCommand}">
                        <i:Interaction.Behaviors>
                            <ib:EventTriggerBehavior EventName="Click">
                                <ib:CallMethodAction MethodName="Hide" TargetObject="{Binding $parent[TreeView].ContextFlyout}" />
                            </ib:EventTriggerBehavior>
                        </i:Interaction.Behaviors>
                    </Button>
                </StackPanel>
            </Menu>
        </Flyout>
    </TreeView.ContextFlyout>
</UserControl>
```
